independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -300,6 +300,33 @@ boolean hasDefaultDestroy() {
         return bound.isEmpty() ? Collections.emptyList() : bound.stream().distinct().sorted().collect(Collectors.toList());
     }
 
+    /**
+     *
+     * @return an ordered list of all interceptors associated with the bean
+     */
+    List<InterceptorInfo> getBoundInterceptors2() {
```

Comment:
`getBoundInterceptors2()`? I think we should replace the original implementation and use this one in `ComponentsProviderGenerator` instead...

Reply:
Aaaaaa damn me, I committed wrong code. Sorry.

Reply:
Should be OK now, I hope. I will have to watch  more what I'm committing :)